guidance_pipeline/write_data.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"os"
+)
+
+func writeData(filename string, data []byte) error {
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
Could be moved into main, if it was doing more I'd say leave the function but make it accept a `Writer` interface.